keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,426 @@
+     * <pre>
+     * List&lt;Secret&gt; secretVersions = new ArrayList();
+     * secretClient.listSecretVersions("secretName").stream().map(secretClient::getSecret).forEach(secretResponse -&gt;
+     *   secretVersions.add(secretResponse.value()));
+     * </pre>
+     *
+     * @param name The name of the secret.
+     * @return A {@link List} containing {@link SecretBase} of all the versions of the specified secret in the vault. List is empty if secret with {@code name} does not exist in key vault
+     */
+    public List<SecretBase> listSecretVersions(String name) {
+        return  service.getSecretVersions(vaultEndpoint, name, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE)
+            .flatMapMany(this::extractAndFetchSecrets).collectList().block();
+    }
+    /**
+     * Gets attributes of all the secrets given by the {@code nextPageLink} that was retrieved from a call to
+     * {@link SecretAsyncClient#listSecrets()}.
+     *
+     * @param nextPageLink The {@link SecretBasePage#nextLink()} from a previous, successful call to one of the list operations.
+     * @return A stream of {@link SecretBase} from the next page of results.
+     */
+    private Flux<SecretBase> listSecretsNext(String nextPageLink) {
+        return service.getSecrets(vaultEndpoint, nextPageLink, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(this::extractAndFetchSecrets);
+    }
+
+    private Publisher<SecretBase> extractAndFetchSecrets(PagedResponse<SecretBase> page) {
+        return extractAndFetch(page, this::listSecretsNext);
+    }
+
+    /**
+     * Gets attributes of all the secrets given by the {@code nextPageLink} that was retrieved from a call to
+     * {@link SecretAsyncClient#listDeletedSecrets()}.
+     *
+     * @param nextPageLink The {@link com.azure.keyvault.implementation.DeletedSecretPage#nextLink()} from a previous, successful call to one of the list operations.
+     * @return A stream of {@link SecretBase} from the next page of results.
+     */
+    private Flux<DeletedSecret> listDeletedSecretsNext(String nextPageLink) {
+        return service.getDeletedSecrets(vaultEndpoint, nextPageLink, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(this::extractAndFetchDeletedSecrets);
+    }
+
+    private Publisher<DeletedSecret> extractAndFetchDeletedSecrets(PagedResponse<DeletedSecret> page) {
+        return extractAndFetch(page, this::listDeletedSecretsNext);
+    }
+
+    //TODO: Extract this in azure-common ImplUtils and use from there
+    private <T> Publisher<T> extractAndFetch(PagedResponse<T> page, Function<String, Publisher<T>> content) {
+        String nextPageLink = page.nextLink();
+        if (nextPageLink == null) {
+            return Flux.fromIterable(page.items());
+        }
+        return Flux.fromIterable(page.items()).concatWith(content.apply(nextPageLink));
+    }
+}
```

Comment:
Same here - extract out

Reply:
PR is out to update azure core with generic implementation.
Once it gets merged will update this.